lib/cloudregister/registerutils.py

```diff
@@ -790,7 +799,7 @@ def get_smt(cache_refreshed=None):
                     '"%s"' % str((server.get_ipv4(), server.get_ipv6()))
                 )
                 # Assume the new server is in the same domain
-                clean_hosts_file(server.get_FQDN())
+                clean_hosts_file(server.get_FQDN(), server.get_registry_FQDN())
```

Comment:
Ditto, let's not support the registry and repository to exist on different domains.

Reply:
I agree the domain should be the same, however, in https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/d91381c7b8dacb11e99c8a3726d8dc242a09d62b/lib/cloudregister/registerutils.py#L1297 the domain name is used but in this line you are pointing here the FQDN is used

note that `smt.get_domain_name()` != `smt.get_FQDN()`
                  `susecloud.net`                   != `smt-ec2.susecloud.net`
with that into account, with the current implementation of `clean_hosts_file` and using the domain name (`smt.get_domain_name()`) the registry entry does **not** get cleaned up

Reply:
I will adapt the clean_hosts_file method to work _only_ with the domain and works to remove the registry entry as well, unless any objection